From a858fdfbcc0f4b05c87326b363959adf508669e4 Mon Sep 17 00:00:00 2001
From: Tommi Franttila <tommi.franttila@intel.com>
Date: Mon, 5 Jan 2015 10:49:10 +0200
Subject: [PATCH 211/215] atomisp2: zoom factor not preserved over
 streams/pipes recreation

When acceleration binaries are loaded the streams and pipes are
destroyed and recreated. This operation loses digital zoom factor.
To preserve zoom factor it is stored when streaming is not enabled and
parameters update is requested. Actual parameter update is performed
when ISP is restarted and atomisp_streamon is called.

Change-Id: Ie4cfb6d3b2049e14c4fe8338b1787722fdc6244a
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-10325
Signed-off-by: Tommi Franttila <tommi.franttila@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/34906
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Tuukka Toivonen <tuukka.toivonen@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c   | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index d0b103d..cfb690a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -3989,10 +3989,15 @@ int atomisp_digital_zoom(struct atomisp_sub_device *asd, int flag,
 			return -EINVAL;
 
 		zoom = max_zoom - min_t(u32, max_zoom - 1, *value);
-
-		dev_dbg(isp->dev, "%s, zoom: %d\n", __func__, zoom);
 		atomisp_css_set_zoom_factor(asd, zoom);
-		atomisp_css_update_isp_params(asd);
+
+		if (asd->stream_env[ATOMISP_INPUT_STREAM_GENERAL].stream &&
+		    asd->streaming == ATOMISP_DEVICE_STREAMING_ENABLED) {
+			dev_dbg(isp->dev, "%s, zoom: %d\n", __func__, zoom);
+			atomisp_css_update_isp_params(asd);
+		} else {
+			asd->params.css_update_params_needed = true;
+		}
 	}
 
 	return 0;
-- 
1.9.1

